src/contents/ui/main.qml

```diff
@@ -112,15 +112,6 @@ PlasmoidItem {
                 icon: plasmoid.configuration.panelIcon
                 imageUrl: player.artUrl
                 imageRadius: plasmoid.configuration.albumCoverRadius
-                type: {
-                    if (!plasmoid.configuration.useAlbumCoverAsPanelIcon) {
-                        return PanelIcon.Type.Icon;
-                    }
-                    if (plasmoid.configuration.fallbackToIconWhenArtNotAvailable && !player.artUrl) {
-                        return PanelIcon.Type.Icon;
-                    }
-                    return PanelIcon.Type.Image;
-                }
```

Comment:
I would keep the type property here, and I would pass the parameter `fallbackToIconWhenArtNotAvailable` to the `PanelIcon` component, something like this:
```qml
PanelIcon {
    size: compact.controlsSize
    icon: plasmoid.configuration.panelIcon
    imageUrl: player.artUrl
    imageRadius: plasmoid.configuration.albumCoverRadius,
    fallbackToIconWhenImageNotAvailable: plasmoid.configuration.fallbackToIconWhenArtNotAvailable
    type: {
        if (!plasmoid.configuration.useAlbumCoverAsPanelIcon) {
            return PanelIcon.Type.Icon;
        }
        return PanelIcon.Type.Image;
    }
    Layout.alignment : Qt.AlignVCenter | Qt.AlignHCenter
}
```

Reply:
Makes sense, updated.

Reply:
I had a bit of trouble getting the logic right at first, I think because we can't rely on visibility to check whether `imageComponent` or `iconComponent` are visible if the `PanelIcon`. It works now but not sure if I got the logic and can be simplified

Reply:
I think now it's simple enough, it looks good to me.